### ISSUE TITLE
Fail badly on badly formatted dates

### ISSUE THREAD

**Issue by drzraf** [jrvarma/icalcli#5]
```
BEGIN:VEVENT
UID:1
SUMMARY:FOO
DTSTART:20220717
DTEND:20222802
END:VEVENT
```

(`DTEND` is invalid)

```
$ icalcli 
'NoneType' object has no attribute 'upper'
```

The actual trace is:
```
Traceback (most recent call last):
  File "/home/x/.local/lib/python3.8/site-packages/icalcli/icalcli.py", line 1429, in repl
    ecal.AgendaQuery(start=FLAGS.start, end=FLAGS.end,
  File "/home/x/.local/lib/python3.8/site-packages/icalcli/icalcli.py", line 1011, in AgendaQuery
    return self._display_queried_events(start, end)
  File "/home/x/.local/lib/python3.8/site-packages/icalcli/icalcli.py", line 960, in _display_queried_events
    event_list = self._search_for_events(start, end, search, field)
  File "/home/x/.local/lib/python3.8/site-packages/icalcli/icalcli.py", line 937, in _search_for_events
    event_list = [ev for ev in self.backend_interface.events
  File "/home/x/.local/lib/python3.8/site-packages/icalcli/icalcli.py", line 938, in <listcomp>
    if self.event_match(ev, start, end, pattern,
  File "/home/x/.local/lib/python3.8/site-packages/icalcli/icalcli.py", line 905, in event_match
    event_end = self._to_datetime(event.decoded('dtend'))
  File "/home/x/.local/lib/python3.8/site-packages/icalendar/cal.py", line 237, in decoded
    return self._decode(name, value)
  File "/home/x/.local/lib/python3.8/site-packages/icalendar/cal.py", line 220, in _decode
    decoded = types_factory.from_ical(name, value)
  File "/home/x/.local/lib/python3.8/site-packages/icalendar/prop.py", line 1044, in from_ical
    decoded = type_class.from_ical(value)
  File "/home/x/.local/lib/python3.8/site-packages/icalendar/prop.py", line 335, in from_ical
    u = ical.upper()
AttributeError: 'NoneType' object has no attribute 'upper'
```

**Comment by jrvarma** [comment]
I do not think much can be done about this. I am using the `icalendar` package to parse the calendar, and as the stack trace shows, it is `icalendar` that is failing badly while parsing the time.

**Comment by jrvarma** [comment]
Fixed in version 0.9.9
